examples/notes/main.go

```diff
@@ -0,0 +1,204 @@
+	client, err := uplink.NewUplink(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create uplink: %v", err)
+	}
+
+	// figure out the encryption key
+	encryptionKey := deriveEncryptionKey(password)
+
+	options := &uplink.ProjectOptions{}
+	options.Volatile.EncryptionKey = &encryptionKey
+
+	// open the project
+	project, err := client.OpenProject(ctx, *satellite, apiKey, options)
+	if err != nil {
+		client.Close()
+		return nil, fmt.Errorf("unable to open project: %v\n", err)
+	}
+
+	// create a bucket for storing data
+	_, err = project.CreateBucket(ctx, *bucketName, nil)
+	if err != nil { // TODO: check whether bucket already exists
+		project.Close()
+		client.Close()
+		return nil, fmt.Errorf("unable to create bucket: %v\n", err)
+	}
+
+	// Open up the desired Bucket within the Project
+	bucket, err := project.OpenBucket(ctx, *bucketName, &uplink.EncryptionAccess{Key: encryptionKey})
+	if err != nil {
+		project.Close()
+		client.Close()
+		return nil, fmt.Errorf("could not open bucket: %v\n", err)
+	}
+
+	return &Notes{
+		Client:  client,
+		Project: project,
+		Bucket:  bucket,
+	}, nil
+}
+
+func (notes *Notes) Close() error {
+	bucketErr := notes.Bucket.Close()
+	projectErr := notes.Project.Close()
+	clientErr := notes.Client.Close()
+
+	if bucketErr != nil || projectErr != nil || clientErr != nil {
+		return fmt.Errorf("error occurred while closing: %v %v %v", bucketErr, projectErr, clientErr)
+	}
+	return nil
+}
+
+func (notes *Notes) Put(ctx context.Context, key, value string) error {
+	// create a io.Reader for uploading
+	buffer := bytes.NewBuffer([]byte(value))
+	// upload all the data
+	err := notes.Bucket.UploadObject(ctx, key, buffer, nil)
+	if err != nil {
+		return fmt.Errorf("unable to upload: %v", err)
+	}
+	return nil
+}
+
+func (notes *Notes) Get(ctx context.Context, key string) (string, error) {
+	// open the object we want to download
+	object, err := notes.Bucket.OpenObject(ctx, key)
+	if err != nil {
+		return "", fmt.Errorf("unable to open object: %v", err)
+	}
+	defer object.Close()
+
+	// we want the whole thing, so range from 0 to -1
+	stream, err := object.DownloadRange(ctx, 0, -1)
+	if err != nil {
+		return "", fmt.Errorf("unable to start downloading: %v", err)
+	}
+	defer stream.Close()
+
+	// read everything from the stream
+	data, err := ioutil.ReadAll(stream)
+	if err != nil {
+		return "", fmt.Errorf("download failed: %v", err)
+	}
+
+	return string(data), nil
+}
+
+func (notes *Notes) List(ctx context.Context, prefix string) ([]string, error) {
+	list, err := notes.Bucket.ListObjects(ctx, &uplink.ListOptions{
+		Prefix:    prefix,
+		Recursive: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("list objects failed: %v", err)
+	}
+
+	keys := []string{}
+	for _, item := range list.Items {
+		keys = append(keys, item.Path)
+	}
+
+	return keys, nil
+}
+
+func deriveEncryptionKey(password string) storj.Key {
+	var key storj.Key
+	// TODO: use some hashing thing here
```

Comment:
I think we now have `storj.NewKey` in master that should be used for deriving the root encryption key from the human-readable-encryption-key-from-the-config.